api/src/controllers/client/authTokenStatus.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 Touca, Inc. Subject to Apache-2.0 License.
+
+import { Request, Response } from 'express'
+
+import { redisClient } from '../../utils/index.js'
+
+type ReqParams = {
+  token?: string
+}
+
+type ResBody =
+  | { status: 'invalid' }
+  | { status: 'unverified' }
+  | { status: 'verified'; apiKey: string }
+
+export async function clientAuthTokenStatus(
+  req: Request<ReqParams>,
+  res: Response<ResBody>
+) {
+  const apiKey = await redisClient.get(`client_auth_token:${req.params.token}`)
+
+  if (apiKey == null) {
+    return res.send({ status: 'invalid' })
+  }
+
+  if (apiKey == '') {
+    return res.send({ status: 'unverified' })
+  }
+
+  res.send({ status: 'verified', apiKey })
```

Comment:
Do we even need status? Cannot we just do:
```suggestion
  return res.status(200).json({ apiKey })
```